### PR TITLE
feat: per-network static responses for RPC methods

### PR DESCRIPTION
**Pull request by paymog**
## Summary

Adds a per-network `staticResponses` list that short-circuits inbound requests matching a configured `(method, params)` pair — returning a stubbed JSON-RPC body without contacting any upstream.

## Motivation

Some chains deviate from common client assumptions in ways that make specific RPC requests unanswerable or produce inconsistent responses across upstreams. The motivating case: chains whose genesis block is at height `1` instead of `0` have no valid answer for `eth_getBlockByNumber("0x0", false)`. Clients that probe block 0 (for example, indexers that scan from genesis) see upstream errors or divergent results, and the upstream may be flagged as misbehaving by eRPC's health tracking.

## Design

- New config: `networks[].staticResponses: StaticResponseConfig[]` where each entry has `method`, `params`, and `response` (exactly one of `result` or `error`).
- Match: exact method equality + recursive deep-equality on params. The matcher tolerates numeric-type divergence between YAML config (`int`) and JSON request decoding (`float64`), and compares maps by key regardless of declared order.
- Short-circuit inserted in `Network.Forward` after method extraction and before the multiplexer / cache / upstream-selection path. Matched requests never touch any upstream.
- Inbound request `id` is echoed in the response (the stub's stored id is irrelevant).
- New metric: `erpc_network_static_response_served_total{project,network,category}`.
- Docs: new "Static responses" section in `config/projects/networks.mdx`.

Non-goals: no matcher DSL, no hex normalization, no per-upstream scope, no change to internal state pollers.

## Test plan

- [x] `go test ./common/... ./erpc/...` (new tests pass, existing tests unaffected)
- [x] New matcher unit tests cover numeric equivalence, nested objects with reordered keys, slice length mismatch, hex-string case sensitivity, and validation edge cases.
- [x] New integration tests cover match-hit (no upstream calls observed via gock), non-matching params fall through, error-shaped stub returns proper JSON-RPC error, and inbound id echo.

## Notes

- `telemetry/metrics.go` also carries a few incidental re-alignments surfaced by `gofmt` on pre-existing code; these aren't semantically related to the feature but were picked up by `make fmt`.
- TS types in `typescript/config/src/generated.ts` are regenerated via `tygo`.

---

> [!NOTE]
> **Medium Risk**
> Adds a new request short-circuit path in `Network.Forward` that can bypass cache/multiplexing/upstream selection, so misconfiguration or matcher edge cases could change production request behavior for specific method+params pairs.
> 
> **Overview**
> Adds per-network `staticResponses` configuration to return canned JSON-RPC `result` or `error` bodies for exact `(method, params)` matches, bypassing cache, multiplexing, and upstream selection.
> 
> Implements a deep params matcher (numeric-type tolerant, map key-order independent), validates entries at config load time, serves matched responses while echoing the inbound request `id`, and records a new Prometheus counter `erpc_network_static_response_served_total`. Docs and generated TypeScript config types are updated accordingly.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c3c93ec4b732698f18bd773529cef55289905964. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: erpc/networks_static_responses.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	resp := common.NewNormalizedResponse().
+		WithRequest(req).
+		WithJsonRpcResponse(jrr)
+
+	telemetry.CounterHandle(
+		telemetry.MetricNetworkStaticResponseServedTotal,
+		n.projectId, n.Label(), req.Finality(ctx).String(),
+	).Inc()
```
[the remark this answers is not in the document]
Comment (paymog) [reply erpc/networks_static_responses.go:58]: Good catch. Fixed — the metric now passes `method` as the `category` label instead of finality.

File: common/validation.go [hunk 2]
```diff
@@ -1319,6 +1319,35 @@ func (n *NetworkConfig) Validate(c *Config) error {
...
+	if s.Response == nil {
+		return fmt.Errorf("response is required")
+	}
+	hasResult := s.Response.Result != nil
+	hasError := s.Response.Error != nil
+	if hasResult && hasError {
+		return fmt.Errorf("response must set exactly one of result or error, got both")
+	}
+	if !hasResult && !hasError {
+		return fmt.Errorf("response must set exactly one of result or error, got neither")
```
[the remark this answers is not in the document]
Comment (paymog) [reply common/validation.go:1346]: Acknowledged — this is a real limitation. Representing a JSON-RPC `null` result in YAML requires distinguishing an explicit-null `Response.Result` from an unset one (e.g. switching `Result` to `*json.RawMessage` or tracking presence via `UnmarshalYAML`). We are intentionally deferring this: the use case (stubbing things like `eth_getTransactionReceipt` for unknown txns) is not what this feature is aimed at, and operators who need a null-result stub can encode it as a JSON-RPC error today. Revisit if a concrete use case comes up.
